vt-digital-libraries-platform/aws-batch-iiif-generator
# Wlh 20240117 refactor

**Pulls updated docker image. Consolidate environment variables. Add logging to help debug**
* * *

**Asana Ticket**: ([link](https://app.asana.com/0/1206697520698812/1206697305609588)) (:star:)

* Other Relevant Links (Meeting note, project page, related pull requests, etc.)

# What does this Pull Request do? (:star:)
Updates docker image. Consolidate environment variables. Add logging to help debug

# What's the changes? (:star:)
A in-depth description of the changes made by this PR. Technical details and possible side effects.
* Updates CF template to point to refactored docker image and to use newer Python runtime. 
* Consolidate and rename environment variables for clarification. 
* Update index.py to add logging to help debug

# How should this be tested?
* Deploy stack using "Launch Stack" button in [README](https://github.com/vt-digital-libraries-platform/aws-batch-iiif-generator/blob/wlh_20240117_refactor/README.md)
* Check that necessary src files specified in [task.json](https://github.com/vt-digital-libraries-platform/aws-batch-iiif-generator/blob/wlh_20240117_refactor/examples/task.json) exist in `AWS_SRC_BUCKET` (If they do not exist you can upload the [sample dataset](https://github.com/vt-digital-libraries-platform/aws-batch-iiif-generator/blob/wlh_20240117_refactor/examples/sample_dataset.zip) to `AWS_SRC_BUCKET` within the "iawa" top level namespace.
* Check that derivates to be generated to NOT already exist in `AWS_DEST_BUCKET`
* Upload [task.json](https://github.com/vt-digital-libraries-platform/aws-batch-iiif-generator/blob/wlh_20240117_refactor/examples/task.json) to `S3BucketName`
* Check that job is created and runs in Batch
* Upon job completion check that derivatives have been created in `AWS_DEST_BUCKET` with the appropriate paths.

# Additional Notes:
Any additional information that you think would be helpful when reviewing this PR.
* branch: `wlh_20240117_refactor`

# Interested parties
@otokama 

(:star:) Required fields
